// clarifai-nodejs-grpc/proto/clarifai/commands/commands_pb.js
// source: proto/clarifai/commands/commands.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');
goog.object.extend(proto, google_protobuf_timestamp_pb);
goog.exportSymbol('proto.clarifai.commands.Commands', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.clarifai.commands.Commands = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.clarifai.commands.Commands, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.clarifai.commands.Commands.displayName = 'proto.clarifai.commands.Commands';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.clarifai.commands.Commands.prototype.toObject = function(opt_includeInstance) {
  return proto.clarifai.commands.Commands.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.clarifai.commands.Commands} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.clarifai.commands.Commands.toObject = function(includeInstance, msg) {
  var f, obj = {
    discardDataOlderThan: (f = msg.getDiscardDataOlderThan()) && google_protobuf_timestamp_pb.Timestamp.toObject(includeInstance, f),
    timeoutDelay: jspb.Message.getFieldWithDefault(msg, 2, 0),
    uploadInterval: jspb.Message.getFieldWithDefault(msg, 3, 0),
    optOut: jspb.Message.getBooleanFieldWithDefault(msg, 4, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.clarifai.commands.Commands}
 */
proto.clarifai.commands.Commands.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.clarifai.commands.Commands;
  return proto.clarifai.commands.Commands.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.clarifai.commands.Commands} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.clarifai.commands.Commands}
 */
proto.clarifai.commands.Commands.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new google_protobuf_timestamp_pb.Timestamp;
      reader.readMessage(value,google_protobuf_timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setDiscardDataOlderThan(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setTimeoutDelay(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setUploadInterval(value);
      break;
    case 4:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setOptOut(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.clarifai.commands.Commands.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.clarifai.commands.Commands.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.clarifai.commands.Commands} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.clarifai.commands.Commands.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDiscardDataOlderThan();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      google_protobuf_timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getTimeoutDelay();
  if (f !== 0) {
    writer.writeUint32(
      2,
      f
    );
  }
  f = message.getUploadInterval();
  if (f !== 0) {
    writer.writeUint32(
      3,
      f
    );
  }
  f = message.getOptOut();
  if (f) {
    writer.writeBool(
      4,
      f
    );
  }
};


/**
 * optional google.protobuf.Timestamp discard_data_older_than = 1;
 * @return {?proto.google.protobuf.Timestamp}
 */
proto.clarifai.commands.Commands.prototype.getDiscardDataOlderThan = function() {
  return /** @type{?proto.google.protobuf.Timestamp} */ (
    jspb.Message.getWrapperField(this, google_protobuf_timestamp_pb.Timestamp, 1));
};


/**
 * @param {?proto.google.protobuf.Timestamp|undefined} value
 * @return {!proto.clarifai.commands.Commands} returns this
*/
proto.clarifai.commands.Commands.prototype.setDiscardDataOlderThan = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.clarifai.commands.Commands} returns this
 */
proto.clarifai.commands.Commands.prototype.clearDiscardDataOlderThan = function() {
  return this.setDiscardDataOlderThan(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.clarifai.commands.Commands.prototype.hasDiscardDataOlderThan = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional uint32 timeout_delay = 2;
 * @return {number}
 */
proto.clarifai.commands.Commands.prototype.getTimeoutDelay = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.clarifai.commands.Commands} returns this
 */
proto.clarifai.commands.Commands.prototype.setTimeoutDelay = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional uint32 upload_interval = 3;
 * @return {number}
 */
proto.clarifai.commands.Commands.prototype.getUploadInterval = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.clarifai.commands.Commands} returns this
 */
proto.clarifai.commands.Commands.prototype.setUploadInterval = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional bool opt_out = 4;
 * @return {boolean}
 */
proto.clarifai.commands.Commands.prototype.getOptOut = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 4, false));
};


/**
 * @param {boolean} value
 * @return {!proto.clarifai.commands.Commands} returns this
 */
proto.clarifai.commands.Commands.prototype.setOptOut = function(value) {
  return jspb.Message.setProto3BooleanField(this, 4, value);
};


goog.object.extend(exports, proto.clarifai.commands);
